File: Solution with python/November LeetCoding Challenge 2021/540. Single Element in a Sorted Array.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        n = len ( nums )
        l = 0
        r = n-1
        
        while ( l < r ) :
            mid = ( l + r ) >> 1
            id1 = -1
            
            if ( mid-1 >= 0 and nums[mid-1] == nums[mid]) :
                id1 = mid-1
                
            elif ( mid+1 < n and nums[mid+1] == nums[mid] ) :
                id1 = mid
                
            else:
                return nums[mid]
            
            if ( id1 % 2  == 1 ) :
                r = mid-1
            else :
                l = mid+1
                
        return nums[l]
        

File: November LeetCoding Challenge 2021/540. Single Element in a Sorted Array.cpp
class Solution {
public:
    int singleNonDuplicate(vector<int>& nums) {
        int ret = 0;
        
        int l = 0, r = nums.size()-1;
        
        while ( l < r ) {
            int mid = ( l + r ) >> 1;
            int id1, id2;
            if ( mid - 1 >= 0 && nums[mid-1] == nums[mid] ) {
                id1 = mid-1;
                id2 = mid;
            }
            else if ( mid + 1 < nums.size() && nums[mid+1] == nums[mid] ) {
                id1 = mid;
                id2 = mid+1;
            }
            else return nums[mid];
            
            if ( id1 % 2 ) r = mid-1;
            else l = mid+1;
        }
        
        return nums[l];
    }
};